### PR TITLE
[JDBC Driver] Adding keyspace name while creating connection

### PR DESCRIPTION
**Pull request by naveen-nahata**
This change is [Reviewable](https://reviewable.io/reviews/youtube/vitess/1780)

### CODE REVIEW

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessJDBCUrl.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public VitessJDBCUrl(String url, Properties info) throws SQLException {
             throw new SQLException(Constants.SQLExceptionMessages.MALFORMED_URL);
         }
         this.keyspace = m.group(7);
-        this.catalog = m.group(8);
+        this.catalog = (m.group(8) == null ? m.group(7) : m.group(8));;
```
Comment (ashudeep-sharma) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessJDBCUrl.java]: remove ;

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessVTGateManager.java [hunk 2]
```diff
@@ -95,8 +112,8 @@ private static VTGateConn getVtGateConn(String hostname, int port, String userna
      * @param username
```
Comment (ashudeep-sharma) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessVTGateManager.java]: @param: keyspace missing

File: java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java [hunk 3]
```diff
@@ -103,5 +103,25 @@
         Assert.assertEquals("user", vitessJDBCUrl.getUsername());
     }
 
+    @Test public void testWithKeyspaceandCatalog() throws Exception {
+        Properties info = new Properties();
+        System.out.println("I am here");
```
Comment (ashudeep-sharma) [inline java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java]: Remove System.out.println("I am here");

File: java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java [hunk 4]
```diff
@@ -103,5 +103,25 @@
...
+
+    @Test public void testWithKeyspace() throws Exception {
+        Properties info = new Properties();
+        VitessJDBCUrl vitessJDBCUrl =
+            new VitessJDBCUrl("jdbc:vitess://user:password@10.33.17.231:15991/vt_shipment",
+                info);
+        Assert.assertEquals(1, vitessJDBCUrl.getHostInfos().size());
+        Assert.assertEquals("vt_shipment",vitessJDBCUrl.getKeyspace());
+        Assert.assertEquals("vt_shipment",vitessJDBCUrl.getCatalog());
+    }
```
Comment (ashudeep-sharma) [inline java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java]: Is the output @ line 124 and 114 correct, what is the difference? one compare the vitessJDBCUrl.getCatalog() to "shipment" and the other to "vt_shipment"

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessJDBCUrl.java (hunk 1 above)
Comment (naveen-nahata) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessJDBCUrl.java]: Done

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessVTGateManager.java (hunk 2 above)
Comment (naveen-nahata) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessVTGateManager.java]: @param keyspace availble in new changes.

 +     * Create vtGateConn object with given identifier.
+     *
+     * @param hostname
+     * @param port
+     * @param username
+     * @param keyspace
+     * @return
+     */

File: java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java (hunk 3 above)
Comment (naveen-nahata) [inline java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java]: Done.

File: java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java (hunk 4 above)
Comment (naveen-nahata) [inline java/jdbc/src/test/java/com/flipkart/vitess/jdbc/test/VitessJDBCUrlTest.java]: If url have one parameter "keyspace" only. we will take default catalog as keyspace only.

In this case url "jdbc:vitess://user:password@10.33.17.231:15991/vt_shipment" has only keyspace.

**Comment by ashudeep-sharma** [comment]
:lgtm:

<details><summary><i>Previously, naveen-nahata wrote…</i></summary>
> #### \[JDBC Driver\] Adding keyspace name while creating connection
</details>


---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/1780#-:-KK7prM8DfRiTvNaWqQC:1417088721)*

**Comment by enisoc** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/1780)*

**Comment by enisoc** [comment]
LGTM

**Pull request merged**